src/SDKs/Azure.Base/data-plane/Azure.Base/Http/Pipeline/ThreadSafeRandom.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Base.Http.Pipeline
+{
+    internal class ThreadSafeRandom: Random
+    {
+        private readonly Random _random = new Random();
+
+        public override int Next()
+        {
+            lock (_random)
+            {
+                return _random.Next();
+            }
+        }
+
```

Comment:
Would it make sense to make these methods async that way we could use semaphoreslim.WaitAsync rather than lock to avoid blocking threadpool threads?

Reply:
Not in this case, this lock is very short-lived as the code that it protects is very fast. I don't think contention would be hight enough to hurt performance but if it would I'll just change to non-blocking thread-local approach.